### DISCUSSION TITLE
Is it possible to get the AudioContext ?

### DISCUSSION THREAD

**Discussion by cristianvogel** [cables-gl/cables_docs#513] [Answers & Questions]
What's the best way to get the current AudioContext being used by a patch?

Or even better in my case - the last node so I can wire the output of the Patch up to another audio processing Node graph?

I mean , from outside an exported Patch

**Comment by cristianvogel** [comment]
OK, so I have been working to make this happen on my super Cables demo. 

I am assigning the GainNode object in the patch to a var. 

<img width="449" alt="Screenshot 2023-04-02 at 13 58 57" src="https://user-images.githubusercontent.com/12829050/229351343-3d2368cb-1a2d-4a61-a8e1-d69899d858b7.png">

I get the var, and it is apparently existing...

```cablesConnect(): void {
		const patch = get(CablesPatch);
		const cablesGainNode = patch.getVar('CablesGainNode');
		console.log('Audio Node: ', cablesGainNode);

		if (cablesGainNode) {
			// wire the Cables audio node to the Audio Engine
			cablesGainNode.on('change', (node: AudioNode, ctx = this._audioContext) => {
				console.log('Valid node?', node);
				if (ctx) node.connect(ctx.destination);
			});
		}
	}
```

But the console says that this `node` value is always null.... 

```Cables Patch is loaded
AudioEngine.ts:182 Audio Node:  zt {_log: a, _eventCallbacks: {…}, _logName: '', _logEvents: false, _listeners: {…}, …}
AudioEngine.ts:44 Elementary loaded 🔊
AudioEngine.ts:187 Valid node? null
patch.js:1 [core_port] onvaluechanged exception cought TypeError: Cannot read properties of null (reading 'connect')
```

I feel like I could be almost there, but I could do with a bit of help.

**Reply by pandrr** [reply]
try `CABLES.WEBAUDIO.getAudioContext()`

**Reply by cristianvogel** [reply]
Yeah totally. I have that one. But I'm actually trying to get the last node and connect it to another graph, for further processing outside of the patch. 

As you can see above though, it'a not looking quite as simple as getting a watcher on a variable.

**Comment by pandrr** [comment]
in your example it seems AudioNode and "node" is not defined, both should be cablesGainNode i guess ?

**Reply by cristianvogel** [reply]
But doesn't `cablesGainNode.on()` pass that in?

**Reply by pandrr** [reply]
ok i guess i missread your code.

maybe it was set before your code is executed ?
try `CABLES.patch.getVar("CablesGainNode").getValue()`

**Reply by cristianvogel** [reply]
ok ! 

this is working for me 
```ts		// wire the Cables audio node to the Audio Engine
			cablesGainNode.on('change', function (node: AudioNode) {
				console.log('Valid node?', node);
				if (node) node.connect(actx.destination);
			});
```

- seems that there is always some timing issues about how and when vars are available.
